### DISCUSSION TITLE
Next.js App Router & ESlint

### DISCUSSION THREAD

**Discussion by ajaymahadeven** [eslint/eslint#18033] [General]
```
>>  npx eslint nextjs/**
Pages directory cannot be found at /Users/admin/Documents/Git/avian/pages or /Users/admin/Documents/Git/avian/src/pages. If using a custom path, please configure with the `no-html-link-for-pages` rule in your eslint config file.

Oops! Something went wrong! :(

ESLint: 8.55.0

No files matching the pattern "nextjs/node_modules" were found.
Please check for typing mistakes in the pattern.

>>  % npx eslint nextjs/app/**
Pages directory cannot be found at /Users/admin/Documents/Git/avian/pages or /Users/admin/Documents/Git/avian/src/pages. If using a custom path, please configure with the `no-html-link-for-pages` rule in your eslint config file.

Oops! Something went wrong! :(

ESLint: 8.55.0

No files matching the pattern "nextjs/app/empty" were found.
Please check for typing mistakes in the pattern.
```

It appears that I'm encountering issues when attempting to run ESLint on all files within the app router. I would appreciate guidance on the correct procedure. Additionally, I currently execute ESLint on pre-commits, but it seems to only consider modified and open files. Any assistance with resolving these challenges would be greatly appreciated.

**Comment by Rec0iL99** [comment]
Hi @thenameisajay, could you please provide a reproducible example for us to try out?

**Comment by ajaymahadeven** [comment]
Hi @Rec0iL99,

Given that this issue arises from a private codebase, I'm limited in the details I can share. However, I'll provide some insight into the issue and its replication process.

### Codebase Organization (Front-end Only)

The structure of the codebase is as follows:

- `nextjs`
 - `package.json`
 - `app`
 - `components`
 - `eslintrc.json`
 - `features`
 - `lib`
 - `public`
 - `.gitignore`

#### Versions

- **Next.js version:** `14.1.0`
- **ESLint version:** `^8.56.0`

#### ESLint Configuration

```json
{
 "env": {
   "browser": true,
   "es2021": true
 },
 "extends": [
   "eslint:recommended",
   "plugin:@typescript-eslint/recommended",
   "plugin:react/recommended",
   "next/core-web-vitals"
 ],
 "parser": "@typescript-eslint/parser",
 "parserOptions": {
   "ecmaVersion": "latest",
   "sourceType": "module"
 },
 "plugins": ["@typescript-eslint", "react"],
 "rules": {
   "react/react-in-jsx-scope": "off",
   "react/prop-types": "off",
   "@typescript-eslint/explicit-module-boundary-types": "off",
   "@typescript-eslint/no-explicit-any": "warn",
   "@typescript-eslint/no-unused-vars": "warn"
 }
}

**Reply by Rec0iL99** [reply]
It's quite difficult to debug a problem with just the config file :(. You could try creating an example from scratch that produces the same problem on [StackBlitz](https://stackblitz.com/) or a similar platform.
